extension/encoding/jsonencodingextension/extension.go

```diff
@@ -0,0 +1,53 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package jsonencodingextension // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/encoding/jsonencodingextension"
+
+import (
+	"context"
+	"time"
+
+	jsoniter "github.com/json-iterator/go"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+)
+
+var _ plog.Unmarshaler = &jsonExtension{}
+var _ plog.Marshaler = &jsonExtension{}
+
+type jsonExtension struct {
+}
+
+func (e *jsonExtension) MarshalLogs(ld plog.Logs) ([]byte, error) {
+	marshaler := &plog.JSONMarshaler{}
+	return marshaler.MarshalLogs(ld)
```

Comment:
The marshaling doesn't match unmarshaling. Please fix it and add a test covering marshal->unmarshal sequence and validating that the result matches the original logs